aws/resource_aws_datasync_location_smb.go

```diff
@@ -0,0 +1,247 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/datasync"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceAwsDataSyncLocationSmb() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDataSyncLocationSmbCreate,
+		Read:   resourceAwsDataSyncLocationSmbRead,
+		Update: resourceAwsDataSyncLocationSmbUpdate,
+		Delete: resourceAwsDataSyncLocationSmbDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"agent_arns": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"mount_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				// Ignore missing config block (stolen from aws_ecs_service)
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if old == "1" && new == "0" {
+						return true
+					}
+					return false
+				},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"version": {
+							Type:         schema.TypeString,
+							Default:      "AUTOMATIC",
```

Comment:
Can use the [AWS SDK constant](https://github.com/terraform-providers/terraform-provider-aws/blob/f39af7fff691603f5dc5b553c618d1dbdb6ea0df/vendor/github.com/aws/aws-sdk-go/service/datasync/api.go#L6453) here.

Reply:
Also please use `validation.StringInSlice()` to ensure that this attribute has only one of the supported values.